### PR TITLE
Use subdomains for event slugs

### PR DESCRIPTION
**Pull request by jonafato**
Switch to using subdomains (as opposed to URL paths) for event slugs.
This currently has feature parity with the old approach. Still to do:

- [x] Serve Flask-Security pages at the subdomain
- [x] Serve the admin at the subdomain
- [ ] Use the slug to filter content in the admin
- [x] Deduplicate event detection logic between the admin and frontend
- [ ] Serve static assets at the subdomain

Notes:

- This change requires setting the application's server name during
  development.
- Remaining todo items are nice to have but not required.

### CODE REVIEW

**Comment by jonafato** [comment]
This likely replaces #169.

File: docker-compose.yml [hunk 1]
```diff
@@ -17,3 +17,4 @@ web:
     - db
   environment:
     - PYTHONUNBUFFERED=1
+    - SERVER_NAME=docker.local:5000
```
Comment (dirn) [inline docker-compose.yml]: This one is tricky. I don't have a `docker.local`. I have `docker`. Someone running native Docker would have `localhost`. Does it need to be set here?

File: pygotham/admin/__init__.py [hunk 2]
```diff
@@ -3,12 +3,16 @@
...
+import arrow
+from flask import g
 from flask_admin import Admin
 from flask_admin.base import AdminIndexView, MenuLink
 from flask.ext.admin.contrib.sqla import ModelView
 from flask_login import current_user
+from sqlalchemy import or_
 
 from pygotham import factory, filters
+from pygotham.events.models import Event
```
Comment (dirn) [inline pygotham/admin/__init__.py]: Is `events` needed here?

File: pygotham/admin/__init__.py [hunk 3]
```diff
@@ -29,6 +33,32 @@ def create_app(settings_override=None):
     """
     app = factory.create_app(__name__, __path__, settings_override)
 
+    @app.url_defaults
```
Comment (dirn) [inline pygotham/admin/__init__.py]: Does this code match what we use elsewhere? Can we put it somewhere we can import it?

File: docker-compose.yml (hunk 1 above)
Comment (jonafato) [inline docker-compose.yml]: This exact name isn't required, but something like it it. This is due to how cookies get set by browsers. Essentially, we need a server name set and a `.` in that name. See the "More on `SERVER_NAME`" section of Flask's [configuration docs](http://flask.pocoo.org/docs/0.10/config/) for more details.

I'm open to suggestions here.

File: pygotham/admin/__init__.py (hunk 2 above)
Comment (jonafato) [inline pygotham/admin/__init__.py]: Not strictly, no. I'll update this to remove it for consistency with other parts of the code.

File: README.rst [hunk 4]
```diff
@@ -110,7 +115,7 @@ Now you're ready to start your PyGotham server::
 
 You should see the PyGotham site at::
 
-    http://0.0.0.0:5000/<slug>/
+    http://<slug>.0.0.0.0:5000/
```
Comment (dirn) [inline README.rst]: Will this actually work? Wouldn't it be `http://$SERVER_NAME/`?

Comment (jonafato) [inline README.rst]: Nice catch. There were two errors there:
- The server name should not actually include the `slug`.
- As you pointed out, if `SERVER_NAME` is set, the URL must match.
Both fixed and working now.

**Pull request merged**